modules/$.string-pad.js

```diff
@@ -0,0 +1,50 @@
+// http://wiki.ecmascript.org/doku.php?id=strawman:string_padding
+'use strict';
+var $def = require('./$.def')
+  , $ = require('./$')
+  , repeat = require('../fn/string/repeat');
+
+module.exports = function(left) {
+  return function $pad(that, minLength, fillChar) {
+    // 2. Let S be ToString(O).
+    var S = that.toString();
+
+    if ( minLength === undefined ) {
+      // 4. If intMinLength is undefined, return S.
+      return S;
+    } else {
+      // 4. Let intMinLength be ToInteger(minLength).
+      var intMinLength = $.toInteger(minLength);
+    }
+
+    // 5. Let fillLen be the number of characters in S minus intMinLength.
+    var fillLen = intMinLength - S.length;
+
+    if ( fillLen < 0 || fillLen === Infinity ) {
+      // 6. If fillLen < 0, then throw a RangeError exception.
+      // 7. If fillLen is +∞, then throw a RangeError exception.
+      throw new RangeError('Cannot satisfy string length ' +
+                            minLength + ' for string ' + JSON.stringify(S));
```

Comment:
It's just to nice the error message. Eg it looks like:
```
Cannot satisfy string length 10 for string "foobar"
```
instead of
```
Cannot satisfy string length 10 for string foobar
```